### ISSUE TITLE
Not working for private/locked channels

### ISSUE THREAD

**Issue by COLDsteem** [chylex/Discord-History-Tracker#69]
Script is running fine otherwise, but it won't track channels that are private/locked, it skips them as if they were not there.
If private/locked channel is the first in the Category, the script won't go past that channel. It will only track that private/locked channel it started with, and then stop after it reaches the previously saved messages in the channel, it won't go to the next channel.
If private/locked channel is not the first in the Category i.e. one of the subsequent in the Category, scrip will just skip it and go on to the next channel that is not private/locked.

**Comment by COLDsteem** [comment]
Ok, I have just found this comment by the chylex in another Issue thread: https://github.com/chylex/Discord-History-Tracker/issues/65#issuecomment-487114912

It would be great if both icons could be used for detection... regular # icon, and the one with # and a lock icon on it.
This way we could have fully functional script and track all the channels currently, until we can find a better solution that will not brake if they change the interface again.

**Comment by chylex** [comment]
There are icons with locks, icons with warnings signs, maybe some other ones? I'd also suspect that the NSFW confirmation dialog breaks automatic channel switch as well, and a ton of other issues/potential issues with DHT.

It looks like there are better options for archiving messages nowadays, that don't rely on details about how the website is built, so they won't break every time Discord updates the interface.

**Comment by COLDsteem** [comment]
Well, if we can implement any and all icon types known to us, then the script would be that much better...

Care to share those? I mean, are you talking about other scripts, some bots perhaps? Thanks!

**Comment by chylex** [comment]
If you search Discord Chat Exporter from Tyrrrz, that one looks very promising - I haven't tried it myself yet so don't take it as an endorsement, but it's very popular and seems to be much more actively developed than DHT ever will be.

**Comment by COLDsteem** [comment]
Yeah, I know about that one, found it back when I've found DHT, but this one seemed way better. :)
Also, DCE needs user Token to work, so I was not that keen to try it. ;)
I'm sticking with DHT!
Thanks though!

**Comment by COLDsteem** [comment]
This is awesome news! Thanks for fixing this! It's working like a charm, it is correctly detecting locked channels and archiving them.

Can you share what was the workaround, what triggers you've used? And if "Switch to Next Channel now recognizes locked and NSFW channels" solution could be used for detecting a New Category?

I'm asking this now, because of something I've already asked in: 
https://github.com/chylex/Discord-History-Tracker/issues/52

I mean, now that you've figured out how to detect apart channels from voice channels, and locked channels, and nsfw channels, did something change in discord, so Categories could be detected as well?

If it did, it would be useful if there was another another group of radio-buttons with the title "After reaching the end of a Category..." in Settings menu.

Thanks for the great app, and keep up the great work!

**Comment by chylex** [comment]
Discord uses 2 different # icons for text channels, I copied both icons into DHT and compare them. This will break if anything changes the icons (Discord update, another addon/script), and it also won't work with some other types of text channels (such as announcement channels that I only noticed after releasing the update, and whatever other weird stuff Discord has).

Not planning to touch categories, they already break channel detection if they're collapsed. Maybe if a better way is found than literally comparing icons, otherwise the automatic channel switching is and has for a long time been a low priority feature due to how often it breaks with updates.
